local_server/local-server-geoprocessing/README.md

```diff
@@ -1,45 +1,43 @@
+## How it works
 
-1. Add raster data to map using as an `ArcGISTiledLayer`.
-2. Create and run a `LocalServer`.
-    * `LocalServer.INSTANCE` creates a local server
-    * `Server.startAsync()` starts the server asynchronously
-3. Wait for server to be in the `LocalServerStatus.STARTED` state.
-    * `Server.addStatusChangedListener()` fires whenever the running status of the local server has changed.
+1. Create and run a local server with `LocalServer.instance`.
+2. Start the server asynchronously with `Server.startAsync()`.
+3. Wait for server to be in the  `LocalServerStatus.STARTED` state.
+   * Callbacks attached to `Server.addStatusChangedListener()` will invoke whenever the status of the local server has changed.
 4. Start a `LocalGeoprocessingService` and run a `GeoprocessingTask`.
-    * `new LocalGeoprocessingService(Url, ServiceType)`, creates a local geoprocessing service
-    * `LocalGeoprocessingService.startAsync()` starts the geoprocessing service asynchronously
-    * `new GeoprocessingTask(LocalGeoprocessingService.getUrl() + "/Contour")`, creates a geoprocessing task that uses the contour lines tool
-5. Create `GeoprocessingParameters` and add a `GeoprocessingDouble` as a parameter using set interval.
-    * `new GeoprocessingParameters(ExecutionType)`, creates geoprocess parameters
-    * `GeoprocessingParameters.getInputs().put("Interval", new GeoprocessingDouble(double))`, creates a parameter with name `Interval` with the interval set as its value.
-6. Create and start a `GeoprocessingJob` using the parameters from above.
-    * `GeoprocessingTask.createJob(GeoprocessingParameters)`, creates a geoprocessing job
-    * `GeoprocessingJob.start()`, starts job
-7. Add contour lines as an `ArcGISMapImageLayer` to map.
-    * get url from local geoprocessing service, `LocalGeoprocessingService.getUrl()`
-    * get server job id of geoprocessing job, `GeoprocessingJob.getServerJobId()`
-    * replace `GPServer` from url with `MapServer/jobs/jobId`, to get generate contour lines data
-    * create a map image layer from that new url and add that layer to the map
+    1. Instantiate `LocalGeoprocessingService(Url, ServiceType)` to create a local geoprocessing service.
+    2. Invoke `LocalGeoprocessingService.start()` to start the service asynchronously.
+    3. Instantiate `GeoprocessingTask(LocalGeoprocessingService.url() + "/Contour")` to create a geoprocessing task that uses the contour lines tool.
+5. Create an instance of `GeoprocessingParameters` and add a `GeoprocessingDouble` as a parameter using `setInterval`.
+    1. Instantiate `GeoprocessingParameters(ExecutionType)` creates geoprocessing parameters.
+    2. Create a parameter using `inputs.insert("ContourInterval", new GeoprocessingDouble(double))` with name `ContourInterval` and with the interval set as its value.
+    3.  Set the input with the interval value using `gpParams.setInputs(inputs)`.
```

Comment:
looks like we don't use this step in Java, I can't find it in the code?

Reply:
You're right, this step is unnecessary. The action is already dealt with in the point above. I've removed it 👍